modules/mono/csharp_script.cpp

```diff
@@ -2867,7 +2867,11 @@ Ref<Resource> ResourceFormatLoaderCSharpScript::load(const String &p_path, const
 	ERR_FAIL_COND_V_MSG(err != OK, Ref<Resource>(), "Cannot load C# script file '" + p_path + "'.");
 #endif
 
-	scr->set_path(p_original_path);
+	// Only one instance of a C# script is allowed to exist.
+	ERR_FAIL_COND_V_MSG(scr->get_path().is_empty() || scr->get_path() != p_original_path, Ref<Resource>(),
```

Comment:
Sorry, I think I got the condition wrong in my previous comment, it should probably be:

```suggestion
	ERR_FAIL_COND_V_MSG(!scr->get_path().is_empty() && scr->get_path() != p_original_path, Ref<Resource>(),
```

Because I think if the path is empty it's the first time we are loading the script so it's fine to take over. We only want the error to happen when the script already has a path and it's different from the registered canonical path.